src/MC/fix_sgcmc.cpp

```diff
@@ -989,6 +1023,70 @@ void FixSemiGrandCanonicalMC::flipAtomGeneric(int flipAtom, int oldSpecies, int
   changedAtoms[flipAtom] = true;
 }
 
+/*********************************************************************
+ * Calculates the change in energy that swapping the given
+ * atom would produce. This routine uses a per-atom energy calculation
+ *********************************************************************/
+
+double FixSemiGrandCanonicalMC::computeEnergyChangeEatom(int flipAtom, int oldSpecies, int newSpecies)
+{
+  double Eold, Enew, deltaE;
+
+  //  printf("In computeEnergyChangeEatom()\n");
+  // Calculate old atomic energy of selected atom
+
+  Eold = force->pair->compute_atomic_energy(flipAtom, neighborList);
+
+  // calculate the old per-atom energy of neighbors
+
+  int* jlist = neighborList->firstneigh[flipAtom];
+  int jnum = neighborList->numneigh[flipAtom];
+
+  for(int jj = 0; jj < jnum; jj++) {
+    int j = jlist[jj];
+    Eold += force->pair->compute_atomic_energy(j, neighborList);
+  }
+
+  // Calculate new per-atom energy of selected atom
+
+  atom->type[flipAtom] = newSpecies;
+
+  Enew = force->pair->compute_atomic_energy(flipAtom, neighborList);
+
+  // calculate the new per-atom energy of neighbors
+
+  for(int jj = 0; jj < jnum; jj++) {
+    int j = jlist[jj];
+    Enew += force->pair->compute_atomic_energy(j, neighborList);
+  }
+
+  atom->type[flipAtom] = oldSpecies;
+
+  deltaE = Enew - Eold;
+
+  //  printf("In computeEnergyChangeEatom() deltaE = %g\n", deltaE);
```

Comment:
Please remove debugging comments.

Reply:
I removed debugging comments from 3 other files in MANYBODY and MC too, added in a separate commit.